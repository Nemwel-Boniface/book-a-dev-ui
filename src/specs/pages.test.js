import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Routes, Route, BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configStore';
import '@testing-library/jest-dom';
// import Developers from '../pages/developers';
import Reservations from '../pages/reservations';
import Developer from '../pages/developer';
import Login from '../pages/login';
import Signup from '../pages/signup';
import CreateDev from '../pages/create_dev';
import FormReservation from '../pages/reserveForm';

describe('Test developers page', () => {
  it('should display the reservation list', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="reservations" element={<Reservations />} />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should be able to see the developer details pages', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="developers/:developerId" element={<Developer />} />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should be able to see the login pages', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={<Login />} />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should be able to see the login pages', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="registration" element={<Signup />} />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should be able to see the login pages', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="create_dev" element={<CreateDev />} />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should be able to see the login pages', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="create_reservation" element={<FormReservation />} />
          </Routes>
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
