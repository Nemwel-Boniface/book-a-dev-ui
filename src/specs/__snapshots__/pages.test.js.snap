// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test developers page should be able to see the developer details pages 1`] = `null`;

exports[`Test developers page should be able to see the login pages 1`] = `
<div
  className="wrapperForm"
>
  <div
    className="form-section"
  >
    <h1>
      Log In
    </h1>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="field"
        id="signup-email-field"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
      />
      <input
        className="field"
        id="login-password-field"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <button
        className="submit_btn"
        type="submit"
      >
        Log In
      </button>
      <span
        className="not_member"
      >
        Not a member?
         
        <a
          href="/registration"
          onClick={[Function]}
        >
          Register
        </a>
      </span>
    </form>
  </div>
</div>
`;

exports[`Test developers page should be able to see the login pages 2`] = `null`;

exports[`Test developers page should be able to see the login pages 3`] = `null`;

exports[`Test developers page should be able to see the login pages 4`] = `null`;

exports[`Test developers page should display the reservation list 1`] = `null`;
